src/usermod.c

```diff
@@ -804,7 +804,7 @@ static void update_gshadow (void)
 	bool was_admin;
 	bool changed;
 	const struct sgrp *sgrp;
-	struct sgrp *nsgrp;
+	struct sgrp *nsgrp = NULL;
 
 	changed = false;
 
```

Comment:
```

shadow-4.15.0/src/usermod.c:864:3: alloc_fn: Storage is returned from allocation function "__sgr_dup".
shadow-4.15.0/src/usermod.c:864:3: var_assign: Assigning: "nsgrp" = storage returned from "__sgr_dup(sgrp)".
shadow-4.15.0/src/usermod.c:964:1: leaked_storage: Variable "nsgrp" going out of scope leaks the storage it points to.
962|                   free (nsgrp);
963|           }
964|-> }
965|   #endif                                /* SHADOWGRP */
966|
```

Where do those line numbers come from?  Did you apply any patches downstream to this file?

Here's what I see in 4.15.0:

```c
alx@debian:~/src/shadow/stable/shadow/4.15.0$ sed -n '864p' src/usermod.c

alx@debian:~/src/shadow/stable/shadow/4.15.0$ sed -n '962,966p' src/usermod.c
static void grp_update (void)
{
	update_group ();
#ifdef SHADOWGRP
	if (is_shadow_grp) {
```

Please say what line numbers those correspond here, and if relevant, the patches applied downstream.

Reply:
Yes, there are downstream only patches that change the line numbering. The patch that changes the line numbering is https://gitlab.com/redhat/centos-stream/rpms/shadow-utils/-/blob/c10s/shadow-4.15.0-audit-update.patch

Reply:
Ahh, ok, so 864 in your report is line 857 in 4.15.0 (`nsgrp = __sgr_dup (sgrp);`).  Thanks!